File: src/textparameters.go
package pdfjet

// TextParameters holds configuration for text rendering
type TextParameters struct {
	font     *Font
	fontSize float32
	x        float32
	y        float32
	text     string
}

// NewTextParameters creates a new TextParameters with default values
func NewTextParameters() *TextParameters {
	return &TextParameters{
		fontSize: 12.0, // Default font size
		x:        0.0,  // Default X position
		y:        0.0,  // Default Y position
	}
}

// SetFont sets the font and returns the receiver for chaining
func (tp *TextParameters) SetFont(font *Font) *TextParameters {
	tp.font = font
	return tp
}

// SetFontSize sets the font size and returns the receiver for chaining
func (tp *TextParameters) SetFontSize(fontSize float32) *TextParameters {
	tp.fontSize = fontSize
	return tp
}

// SetTextLocation sets the X, Y location and returns the receiver for chaining
func (tp *TextParameters) SetTextLocation(x, y float32) *TextParameters {
	tp.x = x
	tp.y = y
	return tp
}

// SetText sets the text content and returns the receiver for chaining
func (tp *TextParameters) SetText(text string) *TextParameters {
	tp.text = text
	return tp
}

File: net/pdfjet/TextParameters.cs
using System;
using System.Text;

namespace PDFjet.NET {
/// <summary>The font, font size, location and text for Stamp.DrawText.</summary>
public class TextParameters {
    internal Font font;
    internal float fontSize;
    internal float x;
    internal float y;
    internal String text;

    // Constructor to initialize with default values (optional)
    /// <summary>Creates text parameters with a font size of 12, located at (0, 0).</summary>
    public TextParameters() {
        this.fontSize = 12f;    // Default font size
        this.x = 0f;            // Default X
        this.y = 0f;            // Default Y
    }

    // Method to set the font
    /// <summary>Sets the font.</summary>
    public TextParameters SetFont(Font font) {
        this.font = font;
        return this;
    }

    // Method to set the font size
    /// <summary>Sets the font size.</summary>
    public TextParameters SetFontSize(float fontSize) {
        this.fontSize = fontSize;
        return this;
    }

    // Method to set the location (X, Y)
    /// <summary>Sets the location of the text.</summary>
    public TextParameters SetTextLocation(float x, float y) {
        this.x = x;
        this.y = y;
        return this;
    }

    /// <summary>Sets the text.</summary>
    public TextParameters SetText(String text) {
        this.text = text;
        return this;
    }
}   // End of TextParameters.cs
}   // End of namespace PDFjet.NET
